app/forms/work_deposit_pathway.rb

```diff
@@ -72,6 +76,21 @@ def scholarly_works?
       Work::Types.scholarly_works.include?(work_type)
     end
 
+    def pathway_fields(work_type)
+      if Work::Types.scholarly_works.include?(work_type)
+        ScholarlyWorks::DetailsForm.form_fields.union(ScholarlyWorks::PublishForm.form_fields)
+      elsif Work::Types.data_and_code.include?(work_type)
+        DataAndCode::DetailsForm.form_fields.union(DataAndCode::PublishForm.form_fields)
+      elsif Work::Types.grad_culminating_experiences.include?(work_type)
+        GradCulminatingExperiences::DetailsForm.form_fields.union(GradCulminatingExperiences::PublishForm.form_fields)
+        # will need to add instruments once that pathway is merged
+        # elsif Work::Types.instrument.include?(work_type)
+        # Instrument::DetailsForm.form_fields.union(Instrument::PublishForm.form_fields)
+      else
+        General::DetailsForm.form_fields.union(WorkVersionFormBase::COMMON_PUBLISH_FIELDS)
+      end
+    end
```

Comment:
Some metaprogramming might reduce some duplication here:

```
def self.pathway_fields(work_type)
    namespace = %w[ScholarlyWorks DataAndCode GradCulminatingExperiences].find do |ns|
      types = Work::Types.public_send(ns.underscore)
      types.include?(work_type)
    rescue NoMethodError
      false
    end

    if namespace
      details = Object.const_get("#{namespace}::DetailsForm")
      publish = Object.const_get("#{namespace}::PublishForm")
      details.form_fields.union(publish.form_fields)
    else
      General::DetailsForm.form_fields.union(WorkVersionFormBase::COMMON_PUBLISH_FIELDS)
    end
  end
```

Reply:
I had trouble with this exact approach, but I was able to take a similar approach & reduce duplication